mbed_build/_internal/config/config.py

```diff
@@ -3,22 +3,28 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 """Build configuration abstraction layer."""
-from dataclasses import dataclass, field
-from typing import List
+from typing import Any, Dict, List, Optional
+from typing_extensions import TypedDict
 
 from mbed_build._internal.config.config_layer import ConfigLayer
 
 
-@dataclass
-class Config:
+class Setting(TypedDict):
+    """Represents a config setting."""
+
+    help: Optional[str]
+    value: Any
+
+
+class Config(TypedDict):
     """Represents a build configuration."""
 
-    settings: dict = field(default_factory=dict)
+    settings: Dict[str, Setting]
```

Comment:
I'm not sure I like this nested dictionary hierarchy. Is there any way to flatten this out? Possibly by making `Config` a normal `class`?

Reply:
Could you provide an example and describe benefits of the alternative approach?